include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -298,18 +298,14 @@ struct reduce_over_group
         auto __local_idx = __item_id.get_local_id(0);
         auto __group_size = __item_id.get_local_range().size();
 
-        auto __k = 1;
-
-        do
+        for (_Size __k = 1; __k < __group_size; __k <<= 1)
         {
             __dpl_sycl::__group_barrier(__item_id);
-            if (__local_idx % (2 * __k) == 0 && __local_idx + __k < __group_size && __global_idx < __n &&
-                __global_idx + __k < __n)
+            if (__local_idx % (2 * __k) == 0 && __local_idx + __k < __group_size && __global_idx + __k < __n)
```

Comment:
Since `__k` is a power of two, `__local_idx % (2 * __k)` can be computed as `__local_idx & (2 * __k - 1)`. I am not sure if this will have any noticeable benefit but it may be worth trying.